### PR TITLE
RTPS and micro-CDR build system improvements

### PR DESCRIPTION
**Pull request by dagar**
This PR fixes some build system issues with RTPS and microCDR.
 - serialization moved from each uORB topic header/source to separate files (eg <uORB/topics/sensor_baro.h> -> <uORB_microcdr/topics/sensor_baro.h>)
 - cdr serialization only generated for listed topics
 - micro-CDR library only included and built if RTPS is detected and topics specified
 - micrortps build dependencies fixed

@mcharleb @bkueng any issues with doing this? It improves build times and doesn't burden targets like the px4io with generating and building all this stuff the linker just throws away.

I still don't like the way this quietly depends on fastrtpsgen being found. It creates a situation where users can build and test locally, but break once pushed to CI. If fastrtpsgen isn't yet a hard requirement we should leave it out of the default builds and have rtps configurations where not being able to find fastrtpsgen is a fatal error (nuttx_px4fmu-v4_rtps, posix_sitl_rtps, etc)

This has been split out of https://github.com/PX4/Firmware/pull/8043.

### CODE REVIEW

File: msg/templates/uorb_microcdr/microRTPS_client.cpp.template [hunk 1]
```diff
@@ -0,0 +1,221 @@
+@###############################################
```
Comment (bkueng) [inline msg/templates/uorb_microcdr/microRTPS_client.cpp.template]: is `msg/templates/uorb/microRTPS_client.cpp.template` still needed?

File: msg/CMakeLists.txt [hunk 2]
```diff
@@ -178,12 +175,85 @@ add_custom_command(OUTPUT ${uorb_sources}
...
+	set(uorb_headers_microcdr)
+	set(msg_out_path_microcdr ${PX4_BINARY_DIR}/uORB_microcdr/topics)
+	set(uorb_sources_microcdr)
+	set(msg_source_out_path_microcdr ${CMAKE_CURRENT_BINARY_DIR}/topics_microcdr_sources)
+
+	# send topic files
+	set(send_topic_files)
+	foreach(topic ${config_rtps_send_topics})
+		list(APPEND send_topic_files ${PX4_SOURCE_DIR}/msg/${topic}.msg)
+	endforeach()
```
[the remark this answers is not in the document]
Comment (dagar) [reply msg/CMakeLists.txt]: done

File: msg/templates/uorb_microcdr/microRTPS_client.cpp.template (hunk 1 above)
Comment (dagar) [reply msg/templates/uorb_microcdr/microRTPS_client.cpp.template]: nope, deleted

File: msg/CMakeLists.txt [hunk 3]
```diff
@@ -178,12 +175,81 @@ add_custom_command(OUTPUT ${uorb_sources}
...
+	set(uorb_sources_microcdr)
+	set(msg_source_out_path_microcdr ${CMAKE_CURRENT_BINARY_DIR}/topics_microcdr_sources)
+
+	# send topic files
+	set(send_topic_files)
+	foreach(topic ${config_rtps_send_topics})
+		set(msg_file ${PX4_SOURCE_DIR}/msg/${topic}.msg)
+		list(APPEND send_topic_files ${msg_file})
+		get_filename_component(msg ${msg_file} NAME_WE)
+		list(APPEND uorb_headers_microcdr ${msg_out_path_microcdr}/${msg}.h)
```
Comment (bkueng) [inline msg/CMakeLists.txt]: Isn't `${msg}` just the same as `${topic}`?

Comment (dagar) [reply msg/CMakeLists.txt]: right

**Pull request merged**